### ISSUE TITLE
p-s to react on pushes to an upstream branch

### ISSUE THREAD

**Issue by TomasTomecek** [packit-service/packit-service#239]
This request is coming from libguestf family of projects:

https://github.com/libguestfs

Their workflow is to review patches on ML and have a tooling to push changes to the github git repos. This project is not using issues nor pull requests. Hence the project requires to have a trigger on push to a branch instead of reacting to PRs

CC @rwmjones

**Comment by TomasTomecek** [comment]
AI: implement this - test pushes to a branch (configurable), set commit status once the build & tests are done
AI: create an issues if something fails (configurable) - stretch goal

**Comment by TomasTomecek** [comment]
@lachmanfrantisek isn't this done?

**Comment by lachmanfrantisek** [comment]
> @lachmanfrantisek isn't this done?

Not yet.

- [ ] Turn on the events in the production github app
- [ ] Finish the database update. Otherwise, only the baby-sit task will return the responses.

**Comment by TomasTomecek** [comment]
good points, thanks for the update!

**Comment by bocekm** [comment]
Hi guys, how does this look like?

**Comment by lachmanfrantisek** [comment]
@bocekm Sorry, we were very busy with the centos-stream in the last two weeks. I would like to finish that in the next week.

**Comment by lachmanfrantisek** [comment]
Finally, we are there!

The workflow and configuration are the same as for pull-requests. Just use `trigger: commit` in the job configuration. (More details can be found in [the documentation](https://packit.dev/docs/configuration/#supported-jobs).)  
  
```yaml  
- job: copr_build  
  trigger: commit  
  metadata:  
    branch: build-branch  
```  
  
The results are shown as commit status. (You can access it with selecting the branch in the `commits` tab on GitHub.)  
  
---  
  
An example build for new commit to the branch can be found here:  
https://github.com/packit-service/hello-world/commits/build-branch  
  
1. Build triggered. Building SRPM.  
![Screenshot from 2020-06-01 13-13-55](https://user-images.githubusercontent.com/20214043/83407731-a4880b80-a411-11ea-89dd-40e6bfcca71b.png)  
2. SRPM successful. Building RPM in Copr.  
![Screenshot from 2020-06-01 13-17-29](https://user-images.githubusercontent.com/20214043/83407730-a3ef7500-a411-11ea-9cdc-09744f9de035.png)  
3. Build successful. Running tests in Testing Farm.  
![Screenshot from 2020-06-01 14-17-43](https://user-images.githubusercontent.com/20214043/83408406-d5b50b80-a412-11ea-89f7-8aaba3ee6246.png)  
4. Tests passed.  
![Screenshot from 2020-06-01 13-30-18](https://user-images.githubusercontent.com/20214043/83407729-a2be4800-a411-11ea-964f-e2010e5c6f4d.png)

(For screenshots I used the staging instance of packit-as-a-service but it's now live on prod as well.)


You can use custom project name as well:
- https://github.com/packit-service/hello-world/commit/9d15a951d64754a5ebbc777031b70de2d15a21fe
- https://copr.fedorainfracloud.org/coprs/packit/the-build-branch-test/build/1419637/

**Comment by TomasTomecek** [comment]
@lachmanfrantisek I'm so happy we finally have this capability \o/

Franta, that write-up is so fancy! Would you be interested in adding it to docs?

**Comment by lachmanfrantisek** [comment]
I need to calm your happiness, we have a small problem with the job-config matching when you combine more triggers and build configs with test ones (e.g. current state in packit).

I am working on the fix now.

**Comment by bocekm** [comment]
@lachmanfrantisek, I'm not sure if your previous comment relates to this, but the status of the copr build is not updated for the commit. It says _"Pending — Starting"_ event though the build has succeeded.

To be seen for example here: https://github.com/oamg/convert2rhel/commit/2b0d22e90a7de2849301ebfb1373b728aa9d014d

**Comment by lachmanfrantisek** [comment]
> @lachmanfrantisek, I'm not sure if your previous comment relates to this, but the status of the copr build is not updated for the commit. It says _"Pending — Starting"_ event though the build has succeeded.
> 
> To be seen for example here: [oamg/convert2rhel@2b0d22e](https://github.com/oamg/convert2rhel/commit/2b0d22e90a7de2849301ebfb1373b728aa9d014d)

This should not be that problem. (If you hit it, it will not do anything.) The fix (#648) is on stage now.

I don't see anything related in Sentry. Maybe some fedmsg issue. I'll take a look to logs.

**Comment by lachmanfrantisek** [comment]
@bocekm I've reproduced that and created an issue for that:
https://github.com/packit-service/packit-service/issues/650

I'll take a look tomorrow.
